features/promotion/complex_promotions.feature

```diff
@@ -0,0 +1,66 @@
+@promotion
+Feature: Receiving discount based on configured promotion
+    In order to pay decreased amount for my order during promotion
+    As a Customer
+    I want to receive discount for my purchase
+
+    Background:
+        Given the store operates on a single channel in "France"
+        And the store classifies its products as "Jackets", "Trousers" and "Dresses"
+        And the store has a product "Black Sabbath jacket" priced at "€100.00"
+        And it belongs to "Jackets"
+        And the store has a product "Iron Maiden trousers" priced at "€80.00"
+        And it belongs to "Trousers"
+        And the store has a product "Metallica dress" priced at "€50.00"
+        And it belongs to "Dresses"
+        And the store has a product "Rammstein bow tie" priced at "€10.00"
+        And it belongs to "Formal attire"
+        And the store has "DHL" shipping method with "€10.00" fee
+
+    @todo
+    Scenario: Receiving discount on first order
+        Given there is a promotion "First order promotion"
+        And it gives "20%" off customer's 1st order
+        When I add product "Metallica Dress" to the cart
+        Then my cart total should be "€40.00"
+        And my discount should be "-€10.00"
+
+    @todo
+    Scenario: Receiving discount on products from specific taxon if order contains products from another taxon
+        Given there is a promotion "Jacket-trousers pack"
+        And it gives "10%" off every product classified as "Jackets" if order contains any product classified as "Trousers"
+        When I add product "Iron Maiden trousers" to the cart
+        And I add product "Black Sabbath jacket" to the cart
+        Then product "Black Sabbath jacket" price should be decreased by "€10.00"
```

Comment:
@TheMadeleine I appreciate whole semantic review, however I will implement these comments only on newly implemented steps and scenarios/feature description, ok? ;) I think rest of them should be fixed in all scenarios and it's such a big change, it should be done in separate PR.